### PR TITLE
Support boolean type for `key`, `endkey`, and `startkey` in view requests

### LINKED ISSUE #494: Querying a view with a boolean type key raises a CloudantArgumentError
### 1. Steps to reproduce and the simplest code sample possible to demonstrate the issue

Create a cloudant database view, and attempt to query the view using the python-cloudant library as follows:
```
results = database.get_view_result('_design/doc', 'view-name', raw_result=True, key=False)
```
The above code will raise a CloudantArgumentError due to the validation on this line https://github.com/cloudant/python-cloudant/blob/33781463ff85c8bb535bd6ede2049e4d3bec6c4c/src/cloudant/_common_util.py#L192

### 2. What you expected to happen

The expected outcome would be that the view get filtered by the boolean type key instead of raising the error

I removed the validation locally and reinitiated the view query, and the key filtered fine from the CouchDB library end i.e. their library accepted a boolean type for the key argument

### 3. What actually happened

A CloudantArgumentError was raised when trying to filter by a boolean value for the argument key

## Environment details

Python 3.9.2
Database server: IBM Cloudant

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Fixes #494 

## Approach

Allow bool type for `endkey`, `key`, and `startkey`. 

## Schema & API Changes

- No change

## Security and Privacy

- No change

## Testing

Edited the `invalid` tests to use `{'foo': 'bar'}` invalid values instead of booleans, and extended the existing tests to validate boolean types.

## Monitoring and Logging

- No change

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 - [FIXED] Fixed result paging for grouped view queries.
 - [FIXED] Incorrect use of username as account name in `Cloudant.bluemix()`.
 - [IMPROVED] Documented use of None account name and url override for `Cloudant.iam()`.
+- [FIXED] View with a boolean type key.
```
Comment (emlaver) [inline CHANGES.md:8]: What about: 
```
[FIXED] Support boolean type for key(s) in view requests.
```

**Review by emlaver — COMMENTED** [review]
There's a couple of `invalid` tests that should be removed since boolean type is valid.  For example: https://github.com/cloudant/python-cloudant/blob/master/tests/unit/param_translation_tests.py#L298 
and 
https://github.com/cloudant/python-cloudant/blob/master/tests/unit/param_translation_tests.py#L243

We'll also want to modify [existing tests](https://github.com/cloudant/python-cloudant/blob/master/tests/unit/param_translation_tests.py#L193) to validate boolean types.

**Comment by mojito317** [comment]
> There's a couple of `invalid` tests that should be removed since boolean type is valid. For example: https://github.com/cloudant/python-cloudant/blob/master/tests/unit/param_translation_tests.py#L298
> and
> https://github.com/cloudant/python-cloudant/blob/master/tests/unit/param_translation_tests.py#L243
> 
> We'll also want to modify [existing tests](https://github.com/cloudant/python-cloudant/blob/master/tests/unit/param_translation_tests.py#L193) to validate boolean types.

I fixed the tests in cfe48ed then I realized there is a need for a modification in the code to pass the tests: d395a1a.

File: CHANGES.md (hunk 1 above)
Comment (mojito317) [reply CHANGES.md]: I used:
```
[FIXED] Support boolean type for key, endkey, and startkey in view requests.
```

**Comment by mojito317** [comment]
As @ricellis  and I agreed offline I put back the invalid tests with `{'foo': 'bar'}` values that have to be invalid in e76f903.

File: CHANGES.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 - [FIXED] Incorrect use of username as account name in `Cloudant.bluemix()`.
 - [IMPROVED] Documented use of None account name and url override for `Cloudant.iam()`.
 - [FIXED] Use custom encoder (if provided) for all view `key` params not just `keys`.
+- [FIXED] Support boolean type for key, endkey, and startkey in view requests.
```
Comment (ricellis) [inline CHANGES.md:9]: nit formatting on params
```suggestion
- [FIXED] Support boolean type for `key`, `endkey`, and `startkey` in view requests.
```

**Review by ricellis — APPROVED** [review]

**Review by emlaver — APPROVED** [review]

**Comment by mojito317** [comment]
Thank you for the review @emlaver and @ricellis! I'm merging this!

**Pull request merged**